lib/mix/test/mix/task_test.exs

```diff
@@ -237,4 +237,14 @@ defmodule Mix.TaskTest do
   test "shortdoc/1" do
     assert Mix.Task.shortdoc(Mix.Tasks.Hello) == "This is short documentation, see"
   end
+
+  test "requirements/1" do
+    assert Mix.Task.requirements(Mix.Tasks.WithRequirement) == ["help compile"]
+  end
+
+  test "run_requirements/1 is run during task execution" do
+    assert ExUnit.CaptureIO.capture_io(fn ->
+             assert Mix.Task.run("with_requirement") == "Task with requirements"
+           end) =~ "mix compile"
```

Comment:
We also need to test that requirements won't run again if the task has already run:

```suggestion
           end) =~ "mix compile"

    Mix.Task.reenable("help")
    
    assert ExUnit.CaptureIO.capture_io(fn ->
             assert Mix.Task.run("with_requirement") == :noop
           end) == ""
```